minetest/minetest
# A light_source should not override sunlight.

Light sources overwrite the sunlight bank, that leads to strange negative light when a light_source also propagates light.
Overwriting the sunlight bank also has not visual purpose.

- Goal of the PR
Correct light propagation for light sources.
This is a corner case when a block propagates light and is a light_source.

- How does the PR work?
Make sure that a lightsource does not override any natural light (i.e. sunlight).

- Does it resolve any reported issue?
Nope

- Does this relate to a goal in [the roadmap](../doc/direction.md)?
N/A

- If not a bug fix, why is this PR needed? What usecases does it solve?

## How to test
1. Make sure light_sources (torches, etc) still work.
2. Create a block that is both a light source and propagates lights. Note that the before this change you could have a negative sunlight-source, a low level light source would actually "suck away" light. A simple way to do this is to just make water_source a light_source.
